lib/src/interface/local_process_manager.dart

```diff
@@ -99,7 +98,79 @@ class LocalProcessManager implements ProcessManager {
   }
 
   @override
+  bool canRun(@checked Object executable, {String workingDirectory}) =>
+      getExecutablePath(executable, workingDirectory) != null;
+
+  @override
   bool killPid(int pid, [ProcessSignal signal = ProcessSignal.SIGTERM]) {
     return Process.killPid(pid, signal);
   }
 }
+
+String _getExecutable(
+    List<dynamic> command, String workingDirectory, bool runInShell) {
+  String commandName = command.first.toString();
+  if (runInShell) {
+    return commandName;
+  }
+  String exe = getExecutablePath(commandName, workingDirectory);
+  if (exe == null) {
+    throw new ArgumentError('Cannot find executable for $commandName.');
+  }
+  return exe;
+}
+
+List<String> _getArguments(List<dynamic> command) =>
+    command.skip(1).map((dynamic element) => element.toString()).toList();
+
+/// Searches the `PATH` for the actual executable that [commandName] is supposed
+/// to launch.
+///
+/// Return `null` if the executable cannot be found.
+@visibleForTesting
+String getExecutablePath(String commandName, String workingDirectory,
+    {bool isWindows, List<String> path, List<String> pathExt}) {
+  workingDirectory ??= Directory.current.path;
+  isWindows ??= Platform.isWindows;
+  String pathSeparator = isWindows ? ';' : ':';
+
+  List<String> extensions = <String>[];
+  if (isWindows && p.extension(commandName).isEmpty) {
+    extensions =
+        pathExt ?? Platform.environment['PATHEXT'].split(pathSeparator);
+  }
+
+  List<String> candidates = <String>[];
+  if (commandName.contains(p.separator)) {
+    candidates =
+        _getCandidatePaths(commandName, <String>[workingDirectory], extensions);
+  } else {
+    List<String> searchPath =
+        path ?? Platform.environment['PATH'].split(pathSeparator);
+    candidates = _getCandidatePaths(commandName, searchPath, extensions);
+  }
+  return candidates.firstWhere((String path) => new File(path).existsSync(),
+      orElse: () => null);
+}
+
+/// Returns all possible combinations of `$searchPath\$commandName.$ext` for
+/// `searchPath` in [searchPaths] and `ext` in [extensions].
+///
+/// If [extensions] is empty, it will just enumerate all
+/// `$searchPath\$commandName`.
+/// If [commandName] is an absolute path, it will just enumerate
+/// `$commandName.$ext`.
+Iterable<String> _getCandidatePaths(
+    String commandName, List<String> searchPaths, List<String> extensions) {
+  List<String> withExtensions = extensions.isNotEmpty
+      ? extensions.map((String ext) => '$commandName$ext').toList()
```

Comment:
That wouldn't work in Windows. In flutter, we do have bin/flutter and bin/flutter.bat, but only the later is executable on Windows. As far as I can tell, windows doesn't have a concept of executables without extensions.